plugins/openai_plugin.py

```diff
@@ -0,0 +1,229 @@
+
+        result.start_time = time.time()
+
+        headers = {"Content-Type": "application/json"}
+
+        data = {
+            "prompt": query["text"],
+            "max_tokens": query["output_tokens"],
+            "min_tokens": query["output_tokens"],
+            "temperature": 1.0,
+            "top_p": 0.9,
+            "seed": 10,
+        }
+
+        if self.model_name is not None:
+            data["model"] = self.model_name
+
+        response = None
+        try:
+            response = requests.post(self.host, headers=headers, json=data, verify=False)
+            response.raise_for_status()
+        except requests.exceptions.ConnectionError as err:
+            result.end_time = time.time()
+            result.error_text = repr(err)
+            if response is not None:
+                result.error_code = response.status_code
+            logger.exception("Connection error")
+            return result
+        except requests.exceptions.HTTPError as err:
+            result.end_time = time.time()
+            result.error_text = repr(err)
+            if response is not None:
+                result.error_code = response.status_code
+            logger.exception("HTTP error")
+            return result
+
+        result.end_time = time.time()
+
+        logger.debug("Response: %s", json.dumps(response.text))
+
+        try:
+            message = json.loads(response.text)
+            error = message.get("error")
+            if error is None:
+                result.output_text = message["choices"][0]["text"]
+                result.output_tokens = message["usage"]["completion_tokens"]
+                result.input_tokens = message["usage"]["prompt_tokens"]
+                result.stop_reason =  message["choices"][0]["finish_reason"]
+            else:
+                result.error_code = response.status_code
+                result.error_text = error
+                logger.error("Error received in response message: %s", error)
+        except json.JSONDecodeError:
+            logger.exception("Response could not be json decoded: %s", response.text)
+            result.error_text = f"Response could not be json decoded {response.text}"
+        except KeyError:
+            logger.exception("KeyError, unexpected response format: %s", response.text)
+            result.error_text = f"KeyError, unexpected response format: {response.text}"
+
+        # For non-streaming requests we are keeping output_tokens_before_timeout and output_tokens same.
+        result.output_tokens_before_timeout = result.output_tokens
+        result.calculate_results()
+
+        return result
+
+
+    def streaming_request_http(self, query: dict, user_id: int, test_end_time: float):
+        headers = {"Content-Type": "application/json"}
+
+        data = {
+            "prompt": query["text"],
+            "max_tokens": query["output_tokens"],
+            "min_tokens": query["output_tokens"],
+            "temperature": 1.0,
```

Comment:
Does these options need to be hardwired? Also there is the option to do queries without the stream option, on the other hand we only use "stream" right? Across all the plugins?

Reply:
In some cases we may want to test stream and non-streaming. 

I think we should keep max/min tokens set to make sure we are getting consistent numbers that don't vary based on when the model predicts the stop token.

As for temperature, top_p, and seed... no those don't need to be hardcoded, and maybe should be configurable, but I think this is low priority so will leave it to a follow-up PR.